lib/db.js

```diff
@@ -453,16 +496,26 @@ Db.prototype.createCollection = deprecateOptions(
  * @param {Db~resultCallback} [callback] The collection result callback
  * @return {Promise} returns Promise if no callback passed
  */
+const statsSchema = {
+  readPreference: { type: [ReadPreference, 'object', 'string'] },
+  scale: { type: 'number' },
+  session: { type: ClientSession }
+};
 Db.prototype.stats = function(options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
-  options = options || {};
+
+  options = validate(
+    statsSchema,
+    options,
+    {},
+    { readPreference: resolveReadPreference(options, { db: this }) },
+    { optionsValidationLevel: this.s.optionsValidationLevel }
+  );
   // Build command object
   const commandObject = { dbStats: true };
   // Check if we have the scale value
   if (options['scale'] != null) commandObject['scale'] = options['scale'];
```

Comment:
`decorateCommand` should be used

Reply:
I made NODE-1735 to track operations that should use `decorateCommand`.